src/app/manage-templates/page.tsx

```diff
@@ -1,22 +1,72 @@
-import { TemplateCard } from '@/components/cards/TemplateCard'
-import { Stack } from '@mui/material'
+import { TemplateBoard } from './ui/TemplateBoard'
+import { apiUrl } from '@/config'
+import { WorkflowStateResponse } from '@/types/dto/workflowStates.dto'
+import { IAssignee, ITemplate } from '@/types/interfaces'
+import { addTypeToAssignee } from '@/utils/addTypeToAssignee'
+import { ClientSideStateUpdate } from '@/hoc/ClientSideStateUpdate'
+import { createNewTemplate, deleteTemplate, editTemplate } from './actions'
+import { ManageTemplateHeader } from './ui/Header'
+import { NoTemplateLayout } from './ui/NoTemplateLayout'
+
+async function getAllWorkflowStates(token: string): Promise<WorkflowStateResponse[]> {
+  const res = await fetch(`${apiUrl}/api/workflow-states?token=${token}`, {
+    next: { tags: ['getAllWorkflowStates'] },
+  })
+
+  const data = await res.json()
+
+  return data.workflowStates
+}
+
+async function getAssigneeList(token: string): Promise<IAssignee> {
+  const res = await fetch(`${apiUrl}/api/users?token=${token}`, {
+    next: { tags: ['getAssigneeList'], revalidate: 0 },
+  })
+
+  const data = await res.json()
+
+  return data.users
+}
+
+async function getAllTemplates(token: string): Promise<ITemplate[]> {
+  const res = await fetch(`${apiUrl}/api/tasks/templates?token=${token}`, {
+    next: { tags: ['getAllTemplates'] },
+  })
+
+  const templates = await res.json()
+
+  return templates.data
+}
+
+export default async function ManageTemplatesPage({ searchParams }: { searchParams: { token: string } }) {
+  const { token } = searchParams
+
+  const [workflowStates, assignee, templates] = await Promise.all([
+    await getAllWorkflowStates(token),
+    addTypeToAssignee(await getAssigneeList(token)),
+    await getAllTemplates(token),
+  ])
 
-export default async function ManageTemplatesPage() {
   return (
-    <AppMargin size={SizeofAppMargin.LARGE}>
-      <Stack
-        direction="column"
-        py="40px"
-        sx={{
-          width: '60%',
-          margin: '0 auto',
-        }}
-        rowGap={4}
-      >
-        <TemplateCard templateName={'Onboarding'} />
-        <TemplateCard templateName={'Sessions'} />
-      </Stack>
-    </AppMargin>
+    <ClientSideStateUpdate workflowStates={workflowStates} token={token} assignee={assignee} templates={templates}>
+      <AppMargin size={SizeofAppMargin.LARGE}>
+        <ManageTemplateHeader showNewTemplateButton={templates.length > 0} />
+        <TemplateBoard
+          handleCreateTemplate={async (payload) => {
+            'use server'
+            await createNewTemplate(token, payload)
+          }}
+          handleDeleteTemplate={async (templateId) => {
+            'use server'
+            await deleteTemplate(token, templateId)
+          }}
+          handleEditTemplate={async (payload, templateId) => {
+            'use server'
+            await editTemplate(token, templateId, payload)
+          }}
```

Comment:
I tried removing the "use server" directive from here and including the "use server" on the top of the `action.tsx` file. However I am getting the following error: `Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server". Or maybe you meant to call this function rather than return it.`. 

This is weird because even according to the next.js doc we can do this. It says, `You can place the directive at the top of an async function to mark the function as a Server Action, or at the top of a separate file to mark all exports of that file as Server Actions.`. 

Please let me know if you've any solutions on this. Do you think this is a bug in next.js?